### PR TITLE
RL-37: Loan Settings and Loan Setup

### PR DESCRIPTION
**Pull request by jvbreen1**
Implement LoanSettings and LoanSetup endpoints and push them up to the S3 CSV bucket that Snowflake ingests.

### CODE REVIEW

File: src/domain/transforms.ts [hunk 1]
```diff
@@ -1,8 +1,22 @@
...
+import { LoanInfo, LoanSettings, Loan, LoanSetup } from './types';
+import {
+  LoanResult,
+  LoanSettingsResponse,
+  LoanSetupResponse
+} from '../net/types';
 import uuid = require('uuid');
 
-export const transformLoan = (loan: LoanResult): Loan => ({
+const parseLoanProDate = (loanProDate: string) => {
```
Comment (jvbreen1) [inline src/domain/transforms.ts]: Test this function, John

File: src/domain/types.ts [hunk 2]
```diff
@@ -8,13 +14,98 @@ export interface Loan {
...
+
+export interface LoanSettings {
+  id: string;
+  loan_pro_id: number | null;
+  loan_application_id: string | null;
+  till_added: Date;
+}
+
+export interface LoanSetup {
+  id: string;
```
Comment (jvbreen1) [inline src/domain/types.ts]: Go through these and make sure to handle nulls

File: src/net/types.ts [hunk 3]
```diff
@@ -1,106 +1,147 @@
...
   title: string;
   settingsId: number;
   setupId: number;
-  //   insurancePolicyId: string;
-  //   collateralId: number;
-  linkedLoan?: string;
-  modId?: number;
+  insurancePolicyId: string;
+  collateralId: number;
+  linkedLoan: null;
```
Comment (jvbreen1) [inline src/net/types.ts]: Well that's not right..

File: src/process-loans.ts [hunk 4]
```diff
@@ -1,18 +1,57 @@
...
+  stageBucketName: string
+) => {
   logger.debug('Getting some loans!');
-  const loans = await loanProClient.getLoans([], '/Loans');
+  const loans = await loanProClient.getLoans(
+    [],
+    '/Loans?$expand=LoanSettings/CustomFieldValues,LoanSetup'
+  );
+
+  // This is kind of gross and non-performant, but I'm ok with that for now since this doesn't need to run in real-time
```
Comment (jvbreen1) [inline src/process-loans.ts]: This should probably be a helper function

File: src/process-loans.ts [hunk 5]
```diff
@@ -1,20 +1,69 @@
...
 const logger = bunyan.createLogger({ name: 'handler' });
 
-export const processLoans = async (loanProClient: LoanProClient) => {
-  logger.debug('Getting some loans!');
-  const loans = await loanProClient.getLoans([], '/Loans');
+interface FlattenedLoans {
+  loanInfo: LoanInfo[];
+  loanSettings: LoanSettings[];
+  loanSetup: LoanSetup[];
+}
```
Comment (recursivefunk) [inline src/process-loans.ts]: What are the rules for typescript, re: semicolons?

Comment (jvbreen1) [reply src/process-loans.ts]: I usually see it like this, with semicolons at the end of each field in a type, interface or class. But the end of the type itself doesn't usually have one. Mostly, I let prettier take care of semicolons for me. I get this if I try to add a semicolon to the end of an interface:

image

**Pull request merged**